ee/src/main/java/org/jboss/as/ee/component/deployers/EECleanUpProcessor.java

```diff
@@ -46,6 +48,7 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
 
     @Override
     public void undeploy(final DeploymentUnit context) {
-
+        final Module module = context.getAttachment(org.jboss.as.server.deployment.Attachments.MODULE);
+        FactoryFinderCache.clearClassLoader(module.getClassLoader());
```

Comment:
This methods is already called in org.wildfly.extension.undertow.deployment.UndertowDeploymentService#stopContext